python_modules/dagster/dagster/_grpc/server.py

```diff
@@ -227,6 +230,14 @@ def __init__(
         self._container_context = check.opt_dict_param(container_context, "container_context")
 
         try:
+            if load_env_vars:
+                # If arguments indicate it wants to load env vars, use the passed-in instance
+                # ref (or the dagster.yaml on the filesystem if no instance ref is provided)
+                with DagsterInstance.from_ref(
+                    instance_ref
+                ) if instance_ref else DagsterInstance.get() as instance:
+                    instance.inject_env_vars(location_name)
```

Comment:
one annoying failure mode this can introduce is other instance config env vars not being present on the user code server - the same type of problem people hit now when they first use the `instance` in sensor/schedule fns